tests/simplex/step-03.cc

```diff
@@ -0,0 +1,297 @@
+
+private:
+  void
+  make_grid_1();
+  void
+  make_grid_2();
+  void
+  setup_system();
+  void
+  assemble_system();
+  void
+  solve();
+  void
+  output_results() const;
+
+  Triangulation<dim, dim> in_triangulation, triangulation;
+  unsigned int            fe_degree;
+  Simplex::FE_P<dim>      fe;
+  Simplex::FE_P<dim>      fe_mapping;
+
+  DoFHandler<dim> dof_handler;
+  MappingFE<dim>  mapping;
+
+  SparsityPattern      sparsity_pattern;
+  SparseMatrix<double> system_matrix;
+
+  Vector<double> solution;
+  Vector<double> system_rhs;
+};
+
+template <int dim>
+Step3<dim>::Step3()
+  : fe_degree(1)
+  , fe(fe_degree)
+  , fe_mapping(1)
+  , dof_handler(triangulation)
+  , mapping(fe_mapping)
+{}
+
+template <int dim>
+void
+Step3<dim>::make_grid_1()
+{
+  unsigned int n_refinements = 3;
+  GridGenerator::subdivided_hyper_cube(in_triangulation,
+                                       2 * std::pow(2, n_refinements));
+  GridGenerator::convert_hypercube_to_simplex_mesh<dim, dim>(in_triangulation,
+                                                             triangulation);
+
+  std::cout << "Number of active cells: " << triangulation.n_active_cells()
+            << std::endl;
+}
+
+template <int dim>
+void
+Step3<dim>::make_grid_2()
+{
+  unsigned int n_refinements = 3;
+  GridGenerator::subdivided_hyper_cube(in_triangulation, 2);
+  GridGenerator::convert_hypercube_to_simplex_mesh<dim, dim>(in_triangulation,
+                                                             triangulation);
+  triangulation.refine_global(n_refinements);
+
+  std::cout << "Number of active cells: " << triangulation.n_active_cells()
+            << std::endl;
+}
+
+template <int dim>
+void
+Step3<dim>::setup_system()
+{
+  dof_handler.distribute_dofs(fe);
+  std::cout << "Number of degrees of freedom: " << dof_handler.n_dofs()
+            << std::endl;
+
+  DynamicSparsityPattern dsp(dof_handler.n_dofs());
+  DoFTools::make_sparsity_pattern(dof_handler, dsp);
+  sparsity_pattern.copy_from(dsp);
+
+  system_matrix.reinit(sparsity_pattern);
+
+  solution.reinit(dof_handler.n_dofs());
+  system_rhs.reinit(dof_handler.n_dofs());
+}
+
+template <int dim>
+void
+Step3<dim>::assemble_system()
+{
+  Simplex::QGauss<dim> quadrature_formula(dim == 2 ? (fe_degree == 1 ? 3 : 7) :
+                                                     (fe_degree == 1 ? 4 : 10));
+
+  FEValues<dim> fe_values(mapping,
+                          fe,
+                          quadrature_formula,
+                          update_values | update_gradients | update_JxW_values);
+
+  const unsigned int dofs_per_cell = fe.n_dofs_per_cell();
+
+  FullMatrix<double> cell_matrix(dofs_per_cell, dofs_per_cell);
+  Vector<double>     cell_rhs(dofs_per_cell);
+
+  std::vector<types::global_dof_index> local_dof_indices(dofs_per_cell);
+
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      // re-compute values &gradients of shape func and determinant at quad -
+      // points
```

Comment:
Comments not needed.

Reply:
holds only for the comments in lines 176 & 177, or for all comments?